program-analysis/echidna/advanced/on-using-cheat-codes.md

```diff
@@ -0,0 +1,53 @@
+# How and when to use cheat codes
+
+**Table of contents:**
+
+- [How and when to use cheat codes](#how-and-when-to-use-cheat-codes)
+  - [Introduction](#introduction)
+  - [Cheat codes available in Echidna](#cheat-codes-available-in-echidna)
+  - [Advise on using cheat codes](#advise-on-using-cheat-codes)
+
+## Introduction
+
+When solidity smart contract testing is performed from Solidity itself, usually requires some "help" in order to tackle some EVM/Solidity limitations. 
```

Comment:
Maybe:
"When testing smart contracts in Solidity itself, it can be helpful to use cheat codes in order to overcome some of the limitations of the EVM/Solidity."

Reply:
Fixed.